tests/migration_test.go

```diff
@@ -932,23 +933,45 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 		Context("with an Fedora shared NFS PVC, cloud init and service account", func() {
 			var pvName string
 			var vmi *v1.VirtualMachineInstance
-			BeforeEach(func() {
+
+			initNFS := func(isIpv6 bool) {
 				tests.SkipNFSTestIfRunnigOnKindInfra()
 				pvName = "test-nfs" + rand.String(48)
 				// Prepare a NFS backed PV
 				By("Starting an NFS POD")
 				os := string(cd.ContainerDiskFedora)
-				nfsIP := tests.CreateNFSTargetPOD(os)
+				nfsIPs := tests.CreateNFSTargetPOD(os)
 				// create a new PV and PVC (PVs can't be reused)
 				By("create a new NFS PV and PVC")
+				nfsIP := libnet.GetIp(libnet.GetPodIpsStrings(nfsIPs), isIpv6)
+				if nfsIP == "" && isIpv6 {
+					Skip("skipping ipv6 nfs test on a single stack cluster")
+				}
+				Expect(nfsIP).NotTo(BeEmpty())
 				tests.CreateNFSPvAndPvc(pvName, "5Gi", nfsIP, os)
-			})
+			}
 
 			AfterEach(func() {
+				By("Deleting the VMI")
+				Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})).To(Succeed())
+
+				By("Waiting for VMI to disappear")
+				tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
```

Comment:
This is just cosmetics, but you could create a function for this.

Reply:
Since I didn't introduce it in this PR, just moved it, let's keep this PR "clean" and do it on a separate one.
This method should be created in utils and use in a lot of places.

Reply:
FYI at `tests.AfterTestsSuiteCleanup` and `tests.BeforeTestCleanup` VMI are deleted by `cleanNamespaces` (no wait for VMI to dispose though)